src/main/java/ir/sahab/dropwizardmetrics/LabelSupportedObjectNameFactory.java

```diff
@@ -45,13 +45,16 @@
      */
     @Override
     public ObjectName createName(String type, String domain, String labeledMetricName) {
-        String labels = extractLabels(labeledMetricName);
+
+        StringBuilder nameBuilder = new StringBuilder(domain.length() + labeledMetricName.length() + 5);
+        nameBuilder.append(quoteDomainIfRequired(domain)).append(':');
         String metricName = extractMetricName(labeledMetricName);
-        String objectName = quoteDomainIfRequired(domain)
-                + ":name=" + metricName
-                + (labels.isEmpty() ? "" : "," + labels);
+        nameBuilder.append("name=").append(quoteValueIfRequired(metricName));
+
+        extractLabels(labeledMetricName, nameBuilder);
```

Comment:
Please refactor it to:

String labels = extractLabels(labeledMetricName);
nameBuidler,append(quoteLabels(labels));